app/assets/javascripts/discourse/app/components/dialog-messages/confirm-session.gjs

```diff
@@ -82,6 +85,32 @@ export default class ConfirmSession extends Component {
     }
   }
 
+  @action
+  async sendPasswordResetEmail(event) {
+    event.preventDefault();
+    try {
+      const result = await ajax("/session/forgot_password.json", {
+        data: { login: this.currentUser.username },
+        type: "POST",
+      });
+
+      if (result.success) {
+        this.errorMessage = null;
+        this.resetEmailSent = I18n.t(
+          "user.confirm_access.password_reset_email_sent"
+        );
+      } else {
+        this.errorMessage = I18n.t(
+          "user.confirm_access.cannot_send_password_reset_email"
+        );
+      }
+    } catch (e) {
+      this.errorMessage = I18n.t(
+        "user.confirm_access.cannot_send_password_reset_email"
+      );
```

Comment:
I'm not sure if this is necessary since it is unlikey the block of code will throw an error.

Reply:
Yes, very unlikely, but server could be down, user may have logged themselves out or made too many requests... we should show something in those cases.

Reply:
But those wouldn’t throw a runtime error would it? The response from the request will just not be successful.

Reply:
The `try/catch` is invoked when the response is successful. And actually, while trying to test this, I started triggering the rate limit errors because we're quite aggressive with limiting "forgot password" requests. Not only do we need to show an error here, but we should wire it up to show the right one, ie: 

<img width="744" alt="image" src="https://github.com/discourse/discourse/assets/368961/61e29fff-87ab-4ac7-8066-0920b971cc5d">

Thanks for the prodding!